app/src/main/java/com/randomappsinc/studentpicker/utils/SpeechUtil.java

```diff
@@ -0,0 +1,45 @@
+package com.randomappsinc.studentpicker.utils;
+
+import android.app.Activity;
+import android.content.ActivityNotFoundException;
+import android.content.Context;
+import android.content.Intent;
+import android.speech.RecognizerIntent;
+
+import com.randomappsinc.studentpicker.R;
+
+import java.util.List;
+import java.util.Locale;
+
+import static android.app.Activity.RESULT_OK;
+
+public class SpeechUtil {
+
+    public static final int SPEECH_REQUEST_CODE = 1;
+
+    public static void openSpeechToTextDialog(Activity activity) {
+        Intent intent = new Intent(RecognizerIntent.ACTION_RECOGNIZE_SPEECH);
+        intent.putExtra(RecognizerIntent.EXTRA_LANGUAGE_MODEL, RecognizerIntent.LANGUAGE_MODEL_FREE_FORM);
+        intent.putExtra(RecognizerIntent.EXTRA_LANGUAGE, Locale.getDefault());
+        intent.putExtra(RecognizerIntent.EXTRA_PROMPT, activity.getString(R.string.list_name_input_speech_message));
+        try {
+            activity.startActivityForResult(intent, SPEECH_REQUEST_CODE);
+            activity.overridePendingTransition(R.anim.slide_in_up, R.anim.slide_out_up);
```

Comment:
I think you misunderstood how this API works similar to me. The first animation refers to how the new activity will animate in. The second refers to how the current activity will animate out. Since the speech dialog goes over the invoking activity, the current activity shouldn't move. This isn't the cleanest solution, but the way I made the current activity stay was to create a "stay.xml" animation like this:

```
<?xml version="1.0" encoding="utf-8"?>
<translate
    xmlns:android="http://schemas.android.com/apk/res/android"
    android:duration="@integer/default_anim_length"
    android:fromYDelta="0%p"
    android:toYDelta="0%p"
    />
```